src/app.js

```diff
@@ -0,0 +1,64 @@
+import TodoCount from './TodoCount.js';
+import TodoList from './TodoList.js';
+import TodoInput from './TodoInput.js';
+import TodoFilter from './TodoFilter.js';
+import {FilterOptions} from './constants.js';
+
+function App() {
+  const $todoInput = document.querySelector('#new-todo-title');
+  const $todoList = document.querySelector('#todo-list');
+  const $todoCount = document.querySelector('.todo-count');
+  const $todoFilter = document.querySelector('.filters');
+  this.data = [];
+  this.filteredData = [];
+  this.selected = 'all';
+
+  this.addItem = (text) => {
+    this.data.push({
+      text: text,
+      isCompleted: false,
+    });
+
+    this.filterItems(this.selected);
+  };
+
+  this.removeItem = (index) => {
+    this.data.splice(index, 1);
+
+    this.filterItems(this.selected);
+  };
```

Comment:
`필터를 구현하고 나니 아이템 삭제/토글 하는 부분에서 버그`에 대해 고민해보았습니다.
@wooooooood 님께서 적어놓았듯이 해야할 일에 필터가 선택되어 있을 때 보여지는 인덱스와 실제 인덱스가 다르기 때문에 버그가 생기는 것 같습니다.
다른 방법도 많이 있겠지만 저 같은 경우 `this.data` 객체에 할일의 고유번호인 id 항목을 추가하여 사용하였습니다.
TodoList의 렌더부분에 data-index대신 `data-id="${id}"`를 사용하여 map에서 나온 인덱스가 아니라 좀 전의 고유아이디를 적용하였는데요. 
`Todolist.js` 29번째줄을 아래와 같이 수정하여 보여지는 순서에 상관없이 할일의 고유아이디를 얻을 수 있습니다. 
```javascript
        if (target.classList.contains('destroy')) {
          const {id} = target.closest('.todo-item').dataset;
          removeItem(id);
        }
```
위에서 얻은 아이디에 해당하는 할일을 삭제하는 기능을 아래와 같이 구현해 보았습니다. 
```suggestion
this.removeItem = (id) => {
  const nextData = this.data.filter((todo) => todo.id.toString() !== id);
  this.data = nextData;
  this.filterItems(this.selected);
};
```
임시로 삭제부분만 예시를 들었는데 토글도 비슷한 방식으로 적용하면 되지 않을까 생각합니다. :frog:

Reply:
아하! index를 재활용(?)하고 싶다는 생각밖에 없었는데 말씀해주신 id 방식을 적용하면 해결될것같습니당! 예시까지 작성해주셔서 감사드립니다~👍✨